util/common/metrics.go

```diff
@@ -27,14 +38,71 @@ func StartSendingMetrics(receiver string, duration, sendingInterval time.Duratio
 			err = SendCollectDMetrics(metricPerInterval, sendingInterval, duration)
 		case "emf":
 			err = SendEMFMetrics(metricPerInterval, metricLogGroup, metricNamespace, sendingInterval, duration)
+		case "app_signals":
+			err = SendAppSignalMetrics() //does app signals have dimension for metric?
+		case "traces":
+			err = SendAppTraceMetrics(duration) //does app signals have dimension for metric?
+
 		default:
 		}
-
 	}()
 
 	return err
 }
 
+func SendAppTraceMetrics(duration time.Duration) error {
+	baseDir := getBaseDir()
+
+	for i := 0; i < int(duration/(5*time.Second)); i++ {
+		startTime := time.Now().UnixNano()
+		traceID := generateTraceID()
+		traceIDStr := hex.EncodeToString(traceID[:])
+
+		err := processTraceFile(filepath.Join(baseDir, "traces.json"), startTime, traceIDStr)
+		if err != nil {
+			fmt.Println("Error processing trace file:", err)
+			return err
+		}
+
+		time.Sleep(5 * time.Second)
+	}
+
+	return nil
+}
+
+func getBaseDir() string {
+	if runtime.GOOS == "windows" {
+		return "C:\\Users\\Administrator\\amazon-cloudwatch-agent-test\\test\\app_signals\\resources\\traces"
+	}
+	return "/Users/ec2-user/amazon-cloudwatch-agent-test/test/app_signals/resources/traces"
+}
+
+func generateTraceID() [16]byte {
```

Comment:
can we use UUID instead?

Reply:
This is the same method we used previously for Linux test, so just keeping it the same as before